pkg/ingester/user_state.go

```diff
@@ -211,11 +210,13 @@ func (m *labelSetCounter) UpdateMetric(ctx context.Context, u *userTSDB, vec *pr
 		for h, entry := range s.valuesCounter {
 			// This limit no longer exists
 			if _, ok := currentLbsLimitHash[h]; !ok {
-				vec.DeleteLabelValues(u.userID, entry.labels.String())
+				metrics.usagePerLabelSet.DeleteLabelValues(u.userID, "max_series", entry.labels.String())
```

Comment:
Small nit. Can we reuse the result of `entry.labels.String()`?
It might be expensive in other implementations in the future like stringlabels where a decode is required.

Reply:
Done!